railties/lib/rails/generators/rails/app/templates/bin/spring.tt

```diff
@@ -1,10 +1,13 @@
 if !defined?(Spring) && [nil, "development", "test"].include?(ENV["RAILS_ENV"])
-  # Load Spring without loading other gems in the Gemfile, for speed.
   gem "bundler"
   require "bundler"
+
+  # Load Spring without loading other gems in the Gemfile, for speed.
   Bundler.locked_gems.specs.find { |spec| spec.name == "spring" }&.tap do |spring|
     Gem.use_paths Gem.dir, Bundler.bundle_path.to_s, *Gem.path
     gem "spring", spring.version
     require "spring/binstub"
+  rescue Gem::MissingSpecError
```

Comment:
https://github.com/rubygems/rubygems/blob/a969c0f16aa56cdb69e2a051f2b093427010e1d9/lib/rubygems/errors.rb#L23-L27

> Instead of rescuing from this class, make sure to rescue from the superclass Gem::LoadError to catch all types of load errors.

```suggestion
  rescue Gem::LoadError
```

Reply:
I've changed it to `Gem::LoadError`, but I'm not 100% sure that is more suitable.  I think the documentation is generally good advice, but, since we are swallowing the error, being more specific might be better.  On the other hand, [`Gem::ConflictError`](https://github.com/rubygems/rubygems/blob/a969c0f16aa56cdb69e2a051f2b093427010e1d9/lib/rubygems/errors.rb#L70-L72) is currently the only other subclass of `Gem::LoadError`, and the `!defined?(Spring)` guard should prevent us from causing / swallowing such an error, so perhaps it doesn't matter.